components/layout/query.rs

```diff
@@ -37,16 +40,112 @@ use style::values::specified::GenericGridTemplateComponent;
 use style::values::specified::box_::DisplayInside;
 use style::values::specified::text::TextTransformCase;
 use style_traits::{ParsingMode, ToCss};
+use webrender_api::units::LayoutTransform;
 
 use crate::ArcRefCell;
+use crate::display_list::StackingContextTree;
 use crate::dom::NodeExt;
 use crate::flow::inline::construct::{TextTransformation, WhitespaceCollapse, capitalize_string};
 use crate::fragment_tree::{
     BoxFragment, Fragment, FragmentFlags, FragmentTree, SpecificLayoutInfo,
 };
+use crate::style_ext::TransformExt;
 use crate::taffy::SpecificTaffyGridInfo;
 
-pub fn process_content_box_request(node: ServoLayoutNode<'_>) -> Option<Rect<Au>> {
+/// Specific helper for post composite query, query that requires the consideration of transform,
+/// mapping of coordinates, scroll offset, etc. It provide some encapsulation of borrowed structures
+/// required in these kind of queries and some useful utilities of it.
+pub(crate) struct PostCompositeQueryHelper<'dom> {
+    scroll_tree: Option<&'dom ScrollTree>,
+}
+
+impl<'dom> PostCompositeQueryHelper<'dom> {
```

Comment:
After a bit of discussion, I think we should remove the `PostCompositeQueryHelper` struct entirely. There is no "composite" step in Servo, so this name is pretty confusing. Instead I think `layout_node_to_root_transform` should probably just do everything here taking the `ScrollTree` as an argument. Sorry for the delay and confusion on getting back to you to you on this.